runtime/vm/JFRChunkWriter.cpp

```diff
@@ -1273,4 +1273,38 @@ VM_JFRChunkWriter::writeThreadDumpEvent()
 
 	return dataStart;
 }
+
+void
+VM_JFRChunkWriter::writeSystemProcessEvent(void *anElement, void *userData)
+{
+	SystemProcessEntry *entry = (SystemProcessEntry *)anElement;
+	VM_JFRChunkWriter *writer = (VM_JFRChunkWriter *)userData;
+	VM_BufferWriter *bufferWriter = writer->_bufferWriter;
+	PORT_ACCESS_FROM_JAVAVM(writer->_vm);
+
+	/* Reserve size field. */
+	U_8 *dataStart = writer->reserveEventSize(bufferWriter);
+
+	/* Write event type. */
+	bufferWriter->writeLEB128(SystemProcessID);
+
+	/* Write start time. */
+	bufferWriter->writeLEB128(entry->ticks);
+
+	/* Write process ID as string. */
+	char pidBuffer[32];
+	snprintf(pidBuffer, sizeof(pidBuffer), "%lu", (unsigned long)entry->pid);
```

Comment:
Missing "a" in "as a string".
This should use `j9str_printf()`, which also makes the cast unnecessary:
```c
	j9str_printf(pidBuffer, sizeof(pidBuffer), "%zu", entry->pid);
```